rxandroid/src/main/java/rx/android/lifecycle/LifecycleHelper.java

```diff
@@ -0,0 +1,92 @@
+package rx.android.lifecycle;
+
+import android.app.Activity;
+import android.app.Application;
+import android.os.Bundle;
+
+import rx.subjects.BehaviorSubject;
+
+/**
+ * This instance lives within the scope of an {@code Activity} in order to forward Lifecycle events
+ * to {@link rx.subjects.BehaviorSubject}
+ *
+ * Thread safe, since it's <i>Thread-confined</i> to the main Thread.
+ *
+ * @author Nabil Hachicha.
+ */
+class LifecycleHelper implements Application.ActivityLifecycleCallbacks {
+    private final Activity activityToMonitor;
+    private final BehaviorSubject<LifecycleEvent> lifecycleSubject;
+    /**
+     * Help track if {@link #onActivityCreated(Activity, Bundle)} was called
+     */
+    private boolean isCreateCalled = false;
```

Comment:
It feels like there ought to be a more consistent way to track this. At the very least you could use `BehaviorSubject.hasValue()` instead of these booleans.